src/java/org/apache/cassandra/locator/AbstractReplicationStrategy.java

```diff
@@ -451,7 +451,7 @@ protected void validateReplicationFactor(String s) throws ConfigurationException
 
     protected void validateExpectedOptions() throws ConfigurationException
     {
-        Collection expectedOptions = recognizedOptions();
+        Collection<String> expectedOptions = recognizedOptions();
```

Comment:
There are some other generics-related warnings in this file, we could fix them here even if it's not strictly related to the patch.

Reply:
Nit: there is a missed unneeded `<String, String>` in line 69.